pkg/apiserver/experiment/experiment.go

```diff
@@ -886,8 +906,30 @@ func (s *Service) listExperiments(c *gin.Context) {
 			if name != "" && chaos.Name != name {
 				continue
 			}
-			if status != "" && chaos.Status != status {
-				continue
+			switch chaos.Kind {
+			case v1alpha1.KindPodChaos:
+				status, err = s.getPodChaosState(chaos.Namespace, chaos.Name, kubeCli)
```

Comment:
`getPodChaosState` and `getIoChaosState` are the same logics. Can we improve these?